packages/core/src/conflict-resolution.ts

```diff
@@ -172,59 +172,51 @@ export class HistoryLog {
  * @param dispatcher - Get commit from IPFS
  * @param cid - Commit CID
  * @param stateLog - Log from the current stream state
- * @param log - Found log so far
+ * @param unappliedCommits - Unapplied commits found so far
  * @param timestamp - Previously found timestamp
  * @private
  */
 export async function fetchLog(
   dispatcher: Dispatcher,
   cid: CID,
   stateLog: HistoryLog,
-  log: LogEntry[] = [],
+  unappliedCommits: CommitData[] = [],
   timestamp?: number
-): Promise<LogEntry[]> {
+): Promise<CommitData[]> {
   if (stateLog.includes(cid)) {
     // already processed
     return []
   }
   const commit = await dispatcher.retrieveCommit(cid)
   if (!commit) throw new Error(`No commit found for CID ${cid.toString()}`)
 
-  let payload = commit
-  const isSignedCommit = StreamUtils.isSignedCommit(commit)
-  if (isSignedCommit) {
-    payload = await dispatcher.retrieveCommit(commit.link)
-    if (!payload) throw new Error(`No commit found for CID ${commit.link.toString()}`)
-  }
-  const prevCid: CID = payload.prev
-  if (!prevCid) {
-    // this is a fake log
-    return []
-  }
-  let nextEntry: LogEntry
-  if (StreamUtils.isAnchorCommit(commit)) {
+  let nextCommitData: CommitData
+  if (StreamUtils.isSignedCommit(commit)) {
+    const linkedCommit = await dispatcher.retrieveCommit(commit.link)
+    if (!linkedCommit) throw new Error(`No commit found for CID ${commit.link.toString()}`)
+    nextCommitData = { cid: cid, type: CommitType.SIGNED, commit: linkedCommit, envelope: commit, timestamp: timestamp }
+  } else if (StreamUtils.isAnchorCommit(commit)) {
     const proof = await dispatcher.retrieveFromIPFS(commit.proof)
     timestamp = proof.blockTimestamp
-    nextEntry = {
-      type: CommitType.ANCHOR,
-      cid: cid,
-      timestamp: timestamp,
-    }
+    nextCommitData = { cid: cid, type: CommitType.ANCHOR, commit: commit, timestamp: timestamp }
   } else {
-    nextEntry = {
-      type: CommitType.SIGNED,
-      cid: cid,
-      timestamp: timestamp,
-    }
+    // For all cases not using DagJWS for signing (e.g. CAIP-10 links)
+    nextCommitData = { cid: cid, type: CommitType.SIGNED, commit: commit, timestamp: timestamp }
```

Comment:
I wonder if we should have a `CommitType.RAW` or `CommitType.UNSIGNED` for these types of commits?  Anyway that's outside the scope of this PR

Reply:
What do you mean by RAW? To me it's very unclear what that word implies here.

Reply:
the same as it does here: https://github.com/ceramicnetwork/js-ceramic/blob/feat/%231421/packages/common/src/stream.ts#L49

We need a name for a commit that isn't an anchor commit but also isn't signed. Like what Caip10Link uses.  We have called them UnsignedCommits in the past, but I don't love defining them by what they're *not* instead of what they *are*. This very PR is about to rename `UnsignedCommit` to `RawCommit`, so that's where I got the idea for `CommitType.RAW`

Reply:
Yeah, don't lover *RawCommit* either. Raw could mean litterally anything, just the raw data. What's the reasoning for not calling it *UnsignedCommit* Seems like it would make sense in the context of this PR?

Fyi, a "raw transaction" in ethereum is a signed transaction encoded as hex. So to me something *raw* could easily mean both signed or unsigned.

Reply:
What about `BasicCommit`?

Reply:
Want me to add this discussion to the renaming ticket I just opened (#1545)? Can mark it `maintenance`.